blog/2024-03-08-structuring-a-large-app.md

```diff
@@ -0,0 +1,391 @@
+---
+author: Masen Furer
+date: 2024-03-08
+title: "Structuring a Large App"
+description: "So your Reflex app is getting large? Here's some advice on how to lay it out."
+image: /blog/structure-large-app.jpg
+---
+
+# First Steps
+
+Reflex makes it easy to get a simple web app running from a single python file:
+
+```python
+import reflex as rx
+
+
+@rx.page()
+def index():
+    return rx.center(
+        rx.heading("Hello Reflex! 👋")
+    )
+
+
+app = rx.App()
+```
+
+However, as the app grows, keeping everything in a single module can quickly
+become messy and unmanageable. Because a Reflex app is just a python project,
+it can be organized into modules and packages (directory with `__init__.py`)
+and use the python import machinery you're likely already familiar with.
+
+# What Makes a Reflex App?
+
+When executing `reflex init` or `reflex run`, the framework will import **two** modules automatically.
+
+## rxconfig.py
```

Comment:
not sure if this is needed in this blog post, it may take focus away from the project structure (since this won't really be changed)

Reply:
personally i think it helps set the stage for how reflex "ingests" a project. rxconfig.py and the main app module are really important pieces, everything else builds on top of that